packages/core/src/components/Dropdown/Dropdown.tsx

```diff
@@ -18,16 +17,284 @@ import {
+  /**
+   * Text to display when there are no options
+   */
+  noOptionsMessage?: (() => string) | typeof NOOP;
+  /**
+   * If set to true, the menu will open when focused
+   */
+  openMenuOnFocus?: boolean;
+  /**
+   * If set to true, the menu will open when clicked
+   */
+  openMenuOnClick?: boolean;
+  /**
+   * If set to true, clear button will be added
+   */
+  clearable?: boolean;
+  /**
+   * custom option render function
+   */
+  optionRenderer?: (option: IComboboxOption) => JSX.Element;
+  /**
+   * custom value render function
+   */
+  valueRenderer?: (...args: unknown[]) => unknown;
+  ValueRenderer?: (...args: unknown[]) => unknown;
+  /**
+   * custom menu render function
+   */
+  menuRenderer?: (...args: unknown[]) => unknown;
+  /**
+   * Default placement of the Dropdown menu in relation to its control. Use "auto" to flip the menu when there isn't enough space below the control.
+   */
+  menuPlacement?: DROPDOWN_MENU_PLACEMENT;
+
+  //  enum() PropTypes.oneOf(Object.values(DROPDOWN_MENU_PLACEMENT)),
+  /**
+   * The CSS position value of the menu, when "fixed" extra layout management might be required
+   * Fixed position can be used to solve the issue of positioning Dropdown inside overflow container like Modal or Dialog
+   */
+  menuPosition?: DROPDOWN_MENU_POSITION;
+  /**
+   * If set to true, the dropdown will be in Right to Left mode
+   */
+  rtl?: boolean;
+  /**
+   * Set default selected value
+   */
+  defaultValue?: DropdownDefaultValue[];
+  /**
+   * The component's value.
+   * When passed, makes this a [controlled](https://reactjs.org/docs/forms.html#controlled-components) component.
+   */
+  value?: DropdownDefaultValue[];
+  /**
+   * Select menu size from `Dropdown.size` - Dropdown.sizes.LARGE | Dropdown.sizes.MEDIUM | Dropdown.sizes.SMALL
+   */
+  size?: BaseSizes;
+  /**
+   * If provided Dropdown will work in async mode. Can be either promise or callback
+   */
+  asyncOptions?: (inputValue: string) => Promise<DropdownDefaultValue[]>;
+  /**
+   * If set to true, fetched async options will be cached
+   */
+  cacheOptions?: boolean;
+  /**
+   * If set, `asyncOptions` will be invoked with its value on mount and the resolved results will be loaded
+   */
+  defaultOptions?: boolean | Record<string, string>[];
+  /**
+   * If set to true, the menu will use virtualization. Virtualized async works only with
+   */
+  isVirtualized?: boolean;
+  /**
+   * Whether the menu should use a portal, and where it should attach
+   */
+  menuPortalTarget?: HTMLElement;
+  /**
+   * Custom function to override existing styles (similar to `react-select`'s `style` prop), for example: `base => ({...base, color: 'red'})`, where `base` is the component's default styles
+   */
+  extraStyles?: (...args: unknown[]) => unknown;
+  /**
+   * Maximum height of the menu before scrolling
+   */
+  maxMenuHeight?: number;
+  /**
+   * Tab index for keyboard navigation purposes
+   */
+  tabIndex?: number;
```

Comment:
We also support strings so it will break existing usages

Reply:
changed to number | string.